CMakeVOL.cmake

```diff
@@ -34,11 +34,31 @@ if (HDF5_VOL_ALLOW_EXTERNAL)
     message (FATAL_ERROR "HDF5_ALLOW_EXTERNAL_SUPPORT must be set to 'GIT' to allow building of external HDF5 VOL connectors")
   endif ()
 
+  set (HDF5_LIB_TARGETS "")
+  set (HDF5_HL_LIB_TARGETS "")
+
+  if (BUILD_STATIC_LIBS)
```

Comment:
If both static and shared are enabled you will have all libraries in the link!

Reply:
Also, make sure these variables aren't used anywhere else, maybe?

Reply:
Fixed in new commit to only add one or the other set of targets. These variable names also aren't used anywhere else in the library.